src/test/kotlin/com/bardsoftware/backend/fleet/rmarkdown/PandocTest.kt

```diff
@@ -53,7 +55,18 @@ class PandocTest {
         val mockConfig = mock<Config> {
             on { getString(CONFIG_KEY) }.thenReturn(CP_COMMAND)
         }
-        val markdownReceiver = MarkdownTaskReceiver(null, tasksDir, publisher, mockConfig)
+
+        val pdf = Paths.get("src", "main", "resources", "example.pdf").toFile()
+        val mockResponse = CompileResponse
+                .newBuilder()
+                .setPdfFile(ByteString.readFrom(pdf.inputStream()))
+                .setStatus(CompileResponse.Status.OK)
+                .build()
+        val mockCompiler = mock<CompilerApi> {
+            on { compile(any()) }.thenReturn(mockResponse)
+        }
+
+        val markdownReceiver = MarkdownTaskReceiver(mockCompiler, tasksDir, publisher, mockConfig)
```

Comment:
So why don't you check at least that you get correct status code in the published result?

Reply:
Isn't it a checking for Ok status in `mockResponse`(line 61)?

Reply:
What do you mean? Yes, you return OK from `CompilerApi` but it is not a guarantee that you get any good code in the published result.